[BUG] Fix Elasticsearch Race Condition in PublisherTest.test_Create_folderpage_Push_page_Remove_folderpush_Create_page_Push_page_SameBundle_shouldbe_created

## Description

The test `PublisherTest.test_Create_folderpage_Push_page_Remove_folderpush_Create_page_Push_page_SameBundle_shouldbe_created` has an **Elasticsearch refresh timing race condition** that causes it to fail when CI resources are contended.

**Root Cause:** The test calls `addContentToIndex()` with `IndexPolicy.FORCE` which synchronously indexes the document and triggers an immediate ES refresh. However, the refresh operation completes asynchronously, and the test immediately queries for the indexed document with `indexCount()`. The ES refresh hasn't finished yet, so the query returns 0 results instead of 1, causing the assertion to fail.

**Technology**: Java (Backend Integration Test)

## Root Cause Analysis

### Failing Code Location
**File:** `dotcms-integration/src/test/java/com/dotcms/publisher/business/PublisherTestUtil.java:268`

```java
// PublisherTestUtil.java:262-270
pageAsset.setIndexPolicy(IndexPolicy.FORCE);
pageAsset.setIndexPolicyDependencies(IndexPolicy.FORCE);
pageAsset.setBoolProperty(Contentlet.IS_TEST_MODE, true);

APILocator.getContentletIndexAPI().addContentToIndex(pageAsset);  // Triggers async refresh
assertEquals(1, APILocator.getContentletAPI()
    .indexCount("+inode:" + pageAsset.getInode() + " " + UUIDGenerator.ulid(), user, false));
    ↑ FAILS HERE: Refresh triggered but not completed yet (~500-1000ms on CI)
```

### Evidence from CI Run 21766524953

**Test failed all 4 retry attempts:**
```
[ERROR] Run 1: expected:<1> but was:<0>  (2.463 s)
[ERROR] Run 2: expected:<1> but was:<0>  (0.810 s)  
[ERROR] Run 3: expected:<1> but was:<0>  (1.785 s)
[ERROR] Run 4: expected:<1> but was:<0>  (0.778 s)
```

**Elasticsearch was healthy and working correctly:**
- Cluster health: GREEN throughout the run
- Indexing DID complete - just 1 second after the test failed
- Test cleanup logs show: "Indexer is not empty, waiting for it to finish" → 1 second later → "Indexer Complete=true"

### What IndexPolicy.FORCE Actually Does

**Code:** `ContentletIndexAPIImpl.java:622-627`
```java
private void indexContentListNow(final List<Contentlet> contentToIndex) {
    final BulkRequest bulkRequest = createBulkRequest(contentToIndex);
    bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);  // Triggers refresh
    putToIndex(bulkRequest);  // Synchronous bulk operation - blocks until indexed
    CacheLocator.getESQueryCache().clearCache();
}
```

**IndexPolicy.FORCE does TWO things:**
1. **Synchronous indexing**: The `bulk()` operation blocks until the document is indexed to ES
2. **Immediate refresh trigger**: Sets `RefreshPolicy.IMMEDIATE` to trigger an ES index refresh

**The timing gap:**
- The `bulk()` call **triggers** an immediate refresh
- BUT Elasticsearch processes refreshes **asynchronously** in the background
- The `bulk()` call returns **before** the refresh fully completes across all shards
- When `indexCount()` executes immediately after, it queries **before** the refresh finishes

**Refresh timing:**
- Fast systems: Refresh completes in 10-100ms
- Under CI load: Refresh can take 500-1000ms
- The test has **zero tolerance** for this delay

## Required Fix (Option 1: Polling)

**Add explicit polling to wait for ES refresh completion:**

```java
// PublisherTestUtil.java:262-270 - UPDATED
pageAsset.setIndexPolicy(IndexPolicy.FORCE);
pageAsset.setIndexPolicyDependencies(IndexPolicy.FORCE);
pageAsset.setBoolProperty(Contentlet.IS_TEST_MODE, true);

APILocator.getContentletIndexAPI().addContentToIndex(pageAsset);

// ✅ ADD THIS: Wait for ES refresh to complete
await().atMost(10, SECONDS)
    .pollInterval(100, MILLISECONDS)
    .until(() -> {
        int count = APILocator.getContentletAPI()
            .indexCount("+inode:" + pageAsset.getInode() + " " + UUIDGenerator.ulid(), 
                       user, false);
        return count == 1;
    });

// Now assertion will succeed
assertEquals(1, APILocator.getContentletAPI()
    .indexCount("+inode:" + pageAsset.getInode() + " " + UUIDGenerator.ulid(), user, false));
```

**Required Import:**
```java
import static org.awaitility.Awaitility.await;
import static java.util.concurrent.TimeUnit.SECONDS;
import static java.util.concurrent.TimeUnit.MILLISECONDS;
```

**Why this fixes it:**
- Polls Elasticsearch every 100ms until document is searchable (max 10 sec)
- Succeeds in <1 second on fast CI runs
- Still passes (taking up to 10 sec) on slow/contended runs
- Fails fast if indexing is truly broken (10 sec timeout with clear error)

## Alternative Fix (Option 2: Use WAIT_FOR Policy)

**Change from FORCE to WAIT_FOR:**

```java
// PublisherTestUtil.java:262-270 - ALTERNATIVE
pageAsset.setIndexPolicy(IndexPolicy.WAIT_FOR);  // Instead of FORCE
pageAsset.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
pageAsset.setBoolProperty(Contentlet.IS_TEST_MODE, true);

APILocator.getContentletIndexAPI().addContentToIndex(pageAsset);  // Blocks until next refresh cycle
assertEquals(1, APILocator.getContentletAPI()
    .indexCount("+inode:" + pageAsset.getInode() + " " + UUIDGenerator.ulid(), user, false));
```

**Why WAIT_FOR might work:**
- Uses `RefreshPolicy.WAIT_UNTIL` which blocks until the next refresh cycle completes
- May add up to 1 second latency (ES default refresh interval)
- More predictable than FORCE but potentially slower

**Recommendation:** Use Option 1 (polling) for better control and faster execution on fast systems.

## Acceptance Criteria

- [x] Root cause identified: ES refresh timing race condition in PublisherTestUtil.java:268
- [ ] Implement await() polling pattern in PublisherTestUtil.createPage()
- [ ] Add required Awaitility imports
- [ ] Verify test passes consistently across 10 consecutive CI runs
- [ ] Verify test fails fast (within 10 sec) if indexing is actually broken
- [ ] Search codebase for similar `addContentToIndex() + indexCount()` patterns that may need the same fix

## Priority

**High** - Blocks merge queue when CI is under load

## Additional Context

**Analysis Documentation:**
- `.claude/diagnostics/run-21766524953/DIAGNOSIS.md` - Full root cause analysis
- CI Run: https://github.com/dotCMS/core/actions/runs/21766524953
- Failing Job: https://github.com/dotCMS/core/actions/runs/21766524953/job/62805244837

**Why This Isn't "Flaky":**
This isn't random flakiness - it's a deterministic race condition. The test has zero tolerance for ES refresh delay, so it fails 100% of the time when ES refresh takes >0ms (which happens under CI load).

**IndexPolicy Values:**
- `DEFER`: Queue for background indexing (async, no guarantee)
- `WAIT_FOR`: Block until next refresh cycle (~1 sec)
- `FORCE`: Trigger immediate refresh but return before completion

**Search for Similar Patterns:**
```bash
# Find other tests with the same race condition pattern
grep -r "addContentToIndex" dotcms-integration/src/test/java | grep -B5 -A5 "indexCount"
```

--- a/dotcms-integration/src/test/java/com/dotcms/publisher/business/PublisherTestUtil.java
+++ b/dotcms-integration/src/test/java/com/dotcms/publisher/business/PublisherTestUtil.java
@@ -254,17 +254,27 @@ public class PublisherTestUtil {
         final HTMLPageAsset pageAsset = (HTMLPageAsset) APILocator.getHTMLPageAssetAPI()
                 .findPage(page.getInode(), sysuser, false);
 
-        pageAsset.setIndexPolicy(IndexPolicy.FORCE);
-        pageAsset.setIndexPolicyDependencies(IndexPolicy.FORCE);
+        pageAsset.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        pageAsset.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
         pageAsset.setBoolProperty(Contentlet.IS_TEST_MODE, true);
 
         APILocator.getContentletAPI().publish(pageAsset, user, false);
 
-        pageAsset.setIndexPolicy(IndexPolicy.FORCE);
-        pageAsset.setIndexPolicyDependencies(IndexPolicy.FORCE);
+        // Use WAIT_FOR policy to block until ES refresh completes and document is searchable.
+        // This fixes the race condition where FORCE would trigger a refresh but return before
+        // completion, causing flaky tests under CI load (see issue #34544).
+        // In a single-node test environment with no replicas, WAIT_FOR should be very fast.
+        pageAsset.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        pageAsset.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
         pageAsset.setBoolProperty(Contentlet.IS_TEST_MODE, true);
 
+        final long startTime = System.currentTimeMillis();
         APILocator.getContentletIndexAPI().addContentToIndex(pageAsset);
+        final long waitTime = System.currentTimeMillis() - startTime;
+        Logger.info(PublisherTestUtil.class,
+                "IndexPolicy.WAIT_FOR completed in " + waitTime + "ms for page inode: "
+                        + pageAsset.getInode());
+
         assertEquals(1, APILocator.getContentletAPI()
                 .indexCount("+inode:" + pageAsset.getInode() + " " + UUIDGenerator.ulid(), user,
                         false));

